modules/nf-commons/src/main/nextflow/extension/FilesEx.groovy

```diff
@@ -1599,4 +1602,20 @@ class FilesEx {
     static String getScheme(Path path) {
         path.getFileSystem().provider().getScheme()
     }
+
+    static String getChecksum(Path path) {
+        if( Files.isDirectory(path) )
+            return null
+
+        if( path instanceof ETagAwareFile )
+            return ((ETagAwareFile)path).getETag()
+
+        final md = MessageDigest.getInstance('MD5')
+        final is = Files.newInputStream(path)
+        final dis = new DigestInputStream(is, md)
+
+        while( dis.read() != -1 ) {}
```

Comment:
With the `DigestInputStream` I have to read through the stream to trigger the md5 computation. This code is reading the stream one byte at a time, which surely is very inefficient. I don't know if the stream is buffering under the hood. Is there a more efficient way to do this? I could also read into a dummy buffer, e.g. 1 MB at a time or something.

Reply:
Likey buffering should speed it up, but I don't know this interface. What chatgpt says!? 😄

Reply:
stackoverflow for the win!

Reply:
Good point, SO recommends ~16KB, so that's what I'll use

https://stackoverflow.com/questions/8748960/how-do-you-decide-what-byte-size-to-use-for-inputstream-read

Reply:
By the way, I asked chatgpt about the optimal buffer size and it told me to run a benchmark. So I asked it to write the benchmark for me. And it did! And it worked! 😆